doc/deploy.rst

```diff
@@ -1,58 +1,53 @@
 Deploy
 ======
 
+New features and bug fixes are continuously added to the develop branch. On
+every merge to master, the version number in ``pypesto/version.py``  should
+be incremented as described below.
 
 Versioning scheme
 -----------------
 
 For version numbers, we use ``A.B.C``, where
 
 * ``C`` is increased for bug fixes,
-* ``B`` is increased for new features,
-* ``A`` is increased for major or API breaking changes.
+* ``B`` is increased for new features and minor API breaking changes,
+* ``A`` is increased for major API breaking changes.
 
 
-Deploy a new release
---------------------
+Creating a new release
+----------------------
 
-When you are done with the changes on your git branch, proceed as follows
-to deploy a new release.
+After new commits have been added to the develop branch, changes can be merged
+to master and a new version of pyPESTO can be released. Every merge to master
+should coincide with an incremented version number and a git tag on the
+respective merge commit.
 
 
 Merge into master
 ~~~~~~~~~~~~~~~~~
 
-First, you need to merge into the master:
+1. create a pull request from develop to master
+2. check that all tests on travis pass
+3. check that the documentation is up-to-date
+4. adapt the version number in the file ``pesto/version.py`` (see above)
+5. update the release notes in ``doc/releasenotes.rst``
+6. request a code review
+7. merge into the origin master branch
 
-1. check that all tests on travis pass
-2. check that the documentation is up-to-date
-3. adapt the version number in the file pesto/version.py
-4. update the release notes in doc/releasenotes.rst
-5. merge into the origin master branch
-
-To be able to actualize perform the merge, sufficient rights may be
+To be able to actually perform the merge, sufficient rights may be
 required. Also, at least one review is required.
 
 
-Upload to PyPI
-~~~~~~~~~~~~~~
-
-After a successful merge, you need to update also the package on PyPI:
-
-5. create a so-called "wheel" via
-
-   ::
-     
-       python setup.py bdist_wheel
-
-   A wheel is essentially a zip archive which contains the source code
-   and the binaries (if any).
-6. upload the archive to PyPI using twine via
-
-   ::
+Creating a release on github
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-       twine upload dist/pypesto-x.y.z-py3-none-any.whl
+After merging into master:
 
-   replacing x.y.z by the latest version number.
+1. create a new release on github
+2. tag the commit with the new version number
```

Comment:
maybe say explicit what is to be done? or link to https://git-scm.com/book/en/v2/Git-Basics-Tagging? E.g. I always forget these commands.

Reply:
No need for `git tag` when creating a release on github. This does it all for you. Will specify the tag format though.